portal/src/graphql/portal/AdminAPIConfigurationScreen.tsx

```diff
@@ -248,7 +320,77 @@ const AdminAPIConfigurationScreenContent: React.VFC<AdminAPIConfigurationScreenC
 const AdminAPIConfigurationScreen: React.VFC =
   function AdminAPIConfigurationScreen() {
     const { appID } = useParams() as { appID: string };
+    const { renderToString } = useContext(Context);
+    const { themes } = useSystemConfig();
     const queryResult = useAppAndSecretConfigQuery(appID);
+    const { updateAppAndSecretConfig, loading: isUpdating } =
+      useUpdateAppAndSecretConfigMutation(appID);
```

Comment:
Handle loading and error properly. See portal/src/graphql/adminapi/UserDetailsConnectedIdentities.tsx:866 for an example.

In short, we have useLoading and useProvideError to put the loading state and the error of a mutation to context, and let the existing architecture to handle them automatically.